### PR TITLE
feat(mock): Add error handling

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
Allow the mock function instance to catch and store errors within the instance.

### CODE REVIEW

File: packages/mock/test/mock.spec.ts [hunk 1]
```diff
@@ -41,8 +41,33 @@ describe('Mock function', () => {
...
     expect(isMock({})).toBeFalsy();
-    expect(isMock({
-      [mockSymbol]: true,
-    })).toBeFalsy();
+    expect(
+      isMock({
+        [mockSymbol]: true,
+      }),
+    ).toBeFalsy();
+  });
```
Comment (dsseng) [inline packages/mock/test/mock.spec.ts:49]: Was this created by our linter during `precommit`?

Comment (ElijahKotyluk) [reply packages/mock/test/mock.spec.ts:49]: Yup, that was pre commit.

Comment (dsseng) [reply packages/mock/test/mock.spec.ts:49]: okay, looks like I messed it up on `master`

Comment (ElijahKotyluk) [reply packages/mock/test/mock.spec.ts:49]: No worries, easy fix.

**Review by dsseng — APPROVED** [review]

**Pull request merged**